src/hotspot/share/gc/g1/g1FullCollector.cpp

```diff
@@ -94,10 +94,15 @@ uint G1FullCollector::calc_active_workers() {
   uint current_active_workers = heap->workers()->active_workers();
   uint active_worker_limit = WorkerPolicy::calc_active_workers(max_worker_count, current_active_workers, 0);
 
+  // Finally consider the amount of used regions.
+  uint regions_per_worker = (uint) (HeapSizePerGCThread / G1HeapRegionSize);
+  uint used_worker_limit = MAX2((heap->num_used_regions() / regions_per_worker), 1u);
```

Comment:
I don't see anything that would ensure regions_per_worker can't be zero, which would lead to divide by zero when calculating used_worker_limit.

Reply:
Good catch, was a bit quick there. But I'm thinking more and more that we should maybe simplify this more to just use, `heap->num_used_regions()` as the limit.

Reply:
That seems like a reasonable idea to me as well.